s3torchbenchmarking/src/s3torchbenchmarking/benchmark_utils.py

```diff
@@ -19,16 +25,67 @@
     nvmlInit()
 
 
+class Distribution:
+    def __init__(self, initial_capacity: int, precision: int = 4):
+        self.initial_capacity = initial_capacity
+        self._values = np.zeros(shape=initial_capacity, dtype=np.float32)
+        self._idx = 0
+        self.precision = precision
+
+    def _expand_if_needed(self):
+        if self._idx > self._values.size - 1:
+            self._values = np.concatenate(
+                self._values, np.zeros(self.initial_capacity, dtype=np.float32)
+            )
+
+    def add(self, val: float):
+        self._expand_if_needed()
+        self._values[self._idx] = val
+        self._idx += 1
+
+    def summarize(self) -> dict:
+        window = self._values[: self._idx]
+        if window.size == 0:
+            return {}
+        return {
+            "n": window.size,
+            "mean": round(float(window.mean()), self.precision),
+            "min": round(np.percentile(window, 0), self.precision),
+            "p50": round(np.percentile(window, 50), self.precision),
+            "p75": round(np.percentile(window, 75), self.precision),
+            "p90": round(np.percentile(window, 90), self.precision),
+            "max": round(np.percentile(window, 100), self.precision),
+        }
+
+    def __str__(self):
+        summary_str = json.dumps(self.summarize(), indent=2)
+        return "Distribution({0})".format(summary_str)
```

Comment:
Not blocking: This looks more like a `__repr__` for the most part, except repr isn't generally multi-line